kernel/packages/shared/apis/PortableExperiences.ts

```diff
@@ -53,13 +41,10 @@ export class PortableExperiences extends ExposableAPI {
   @exposeMethod
   async kill(pid: string): Promise<boolean> {
     const parcelIdentity: ParcelIdentity = this.options.getAPIInstance(ParcelIdentity)
-    const currentExecutor: Executor = { type: ExecutorType.SCENE, identifier: parcelIdentity.cid }
-    if (
-      this.currentPortableExperiences.has(pid) &&
-      JSON.stringify(this.currentPortableExperiences.get(pid)) === JSON.stringify(currentExecutor)
-    ) {
+
+    const portableExperience = getPortableExperience(pid)
+    if (!!portableExperience && portableExperience.cid == parcelIdentity.cid) {
```

Comment:
Shouldn't it be the parent cid?
```suggestion
    if (!!portableExperience && portableExperience.parentCid == parcelIdentity.cid) {
```